src/test/scala/mesosphere/marathon/storage/repository/GcActorTest.scala

```diff
@@ -618,5 +620,39 @@ class GcActorTest extends AkkaUnitTest with TestKitBase with GivenWhenThen with
         noMoreInteractions(deployRepo)
       }
     }
+    "starting up with provided scan interval" should {
+      "start in a resting state" in {
+        val appRepo = mock[AppRepositoryImpl[RamId, String, Identity]]
+        val podRepo = mock[PodRepositoryImpl[RamId, String, Identity]]
+        val groupRepo = mock[StoredGroupRepositoryImpl[RamId, String, Identity]]
+        val deployRepo = mock[DeploymentRepositoryImpl[RamId, String, Identity]]
+        val actor = TestFSMRef(new GcActor(deployRepo, groupRepo, appRepo, podRepo, 25, 32, 100.millis))
+
+        actor.stateName shouldEqual Resting
+      }
+      "go to ReadyForGc when the resting time is over" in {
+        val appRepo = mock[AppRepositoryImpl[RamId, String, Identity]]
+        val podRepo = mock[PodRepositoryImpl[RamId, String, Identity]]
+        val groupRepo = mock[StoredGroupRepositoryImpl[RamId, String, Identity]]
+        val deployRepo = mock[DeploymentRepositoryImpl[RamId, String, Identity]]
+        val actor = TestFSMRef(new GcActor(deployRepo, groupRepo, appRepo, podRepo, 25, 32, 100.millis))
+
+        actor.stateName shouldEqual Resting
+        actor.isTimerActive(ScanIntervalTimerName) shouldEqual true
+        eventually(actor.stateName shouldEqual ReadyForGc)
+        actor.isTimerActive(ScanIntervalTimerName) shouldEqual false
+      }
+      "go to resting state after compaction process is done" in {
+        val appRepo = mock[AppRepositoryImpl[RamId, String, Identity]]
+        val podRepo = mock[PodRepositoryImpl[RamId, String, Identity]]
+        val groupRepo = mock[StoredGroupRepositoryImpl[RamId, String, Identity]]
+        val deployRepo = mock[DeploymentRepositoryImpl[RamId, String, Identity]]
+        val actor = TestFSMRef(new GcActor(deployRepo, groupRepo, appRepo, podRepo, 25, 32, 100.millis))
+
+        actor.setState(Compacting, BlockedEntities())
+        actor ! CompactDone
+        actor.stateName shouldEqual Resting
```

Comment:
If it would be a plain actor, then we'd have a race condition. However, since it's a `TestActorRef`, all interactions are synchronous and there is no space for race conditions.